Zend/tests/return_types/noreturn_disallowed3.phpt

```diff
@@ -0,0 +1,19 @@
+--TEST--
+noreturn return type: unacceptable cases: implicit return
+--FILE--
+<?php
+
+function foo(): noreturn {
+    if (false) {
+        throw new Exception('bad');
+    }
+}
+
+foo();
+?>
+--EXPECTF--
+Fatal error: Uncaught TypeError: foo(): Nothing was expected to be returned in %s:%d
```

Comment:
Hope you don't mind my suggestions:
* Mention `noreturn` explicitly
* Perhaps instead of "Nothing was expected to be returned" something like "Attempt to return from a noreturn function" should be clearer?
* It's not a type error, but rather a logic one. The `noreturn` "type" is merely how functions are marked as never returning.

Reply:
> Perhaps instead of "Nothing was expected to be returned" something like "Attempt to return from a noreturn function" should be clearer?

This error is printed when the function ends without returning. So maybe something like "noreturn function was expected to throw, terminate or run infinitely". That's very explicit, maybe somebody has a better idea.

> The `noreturn` "type" is merely how functions are marked as never returning.

`noreturn` is an actual type in PHPs type system. We could add a different error but I wouldn't say `TypeError` is wrong.